src/test/java/software/sigma/bu003/internship/coparts/client/technomir/client/TechnomirClientTest.java

```diff
@@ -0,0 +1,176 @@
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import software.sigma.bu003.internship.coparts.client.technomir.config.TechnomirApiUri;
+import software.sigma.bu003.internship.coparts.client.technomir.config.TehnomirClientConfig;
+import software.sigma.bu003.internship.coparts.client.technomir.entity.request.TehnomirPayLoad;
+import software.sigma.bu003.internship.coparts.client.technomir.entity.responce.TechnomirPartWrapper;
+import software.sigma.bu003.internship.coparts.client.technomir.entity.responce.StockTechnomirPartWrapper;
+import software.sigma.bu003.internship.coparts.client.technomir.entity.responce.TechnomirPart;
+import software.sigma.bu003.internship.coparts.client.technomir.entity.responce.StockTechnomirPart;
+import software.sigma.bu003.internship.coparts.client.technomir.exception.TechnomirClientException;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class TechnomirClientTest {
+
+    @Mock
+    private RestTemplate restTemplate;
+
+    @Mock
+    private TehnomirClientConfig tehnomirClientConfig;
+
+    @InjectMocks
+    private TechnomirClient SUT;
+
+    private final String URL_TO_API = "http://test:8080";
+
+    private final String PART_CODE = "7701";
+
+    private final TehnomirPayLoad testTehnomirPayLoad = new TehnomirPayLoad();
+
+    private final StockTechnomirPart stockTechnomirPart = new StockTechnomirPart();
+
+    private final StockTechnomirPartWrapper stockTechnomirPartWrapper = new StockTechnomirPartWrapper();
+
+    private final TechnomirPartWrapper technomirPartWrapper = new TechnomirPartWrapper();
+    private final TechnomirPart technomirPart = new TechnomirPart();
+
+
+    @BeforeEach
+    public void init(){
+        stockTechnomirPartWrapper.setData(List.of(stockTechnomirPart));
+        technomirPartWrapper.setData(List.of(technomirPart));
+
+        String API_TOKEN = "12345";
+        testTehnomirPayLoad.setApiToken(API_TOKEN);
+
+        when(tehnomirClientConfig.getUrlToApi()).thenReturn(URL_TO_API);
+        when(tehnomirClientConfig.getApiToken()).thenReturn(API_TOKEN);
+    }
+
+    @Test
+    void shouldReturnListStockPartsIfSuccessfully() {
+        when(restTemplate.postForObject(
+                URL_TO_API + TechnomirApiUri.PRICE_GET_STOCK_PRICE.getStr(),
+                testTehnomirPayLoad,
+                StockTechnomirPartWrapper.class)).thenReturn(stockTechnomirPartWrapper);
+
+        Optional<List<StockTechnomirPart>> actual = SUT.getListStockParts();
+
+        assertEquals(Optional.of(List.of(stockTechnomirPart)), actual);
+        verify(tehnomirClientConfig, times(1)).getUrlToApi();
```

Comment:
you don't need to verfify verify(tehnomirClientConfig, times(1)).getUrlToApi(); it's not  a business logic

Reply:
corrected